liquidator/index.js

```diff
@@ -128,22 +139,43 @@ async function run(logger, empAddress, pollingDelay, priceFeedConfig, liquidator
       });
     }
 
+    // Create a execution loop that will run indefinitely (or yield early if in serverless mode)
     while (true) {
-      // Get the current synthetic balance. Used to define the max tokens to liquidate.
-      const currentSyntheticBalance = await syntheticToken.balanceOf(accounts[0]);
-      // Update the liquidators state. This will update the clients, price feeds and gas estimator.
-      await liquidator.update();
-      // Check for liquidatable positions and submit liquidations. Bounded by current synthetic balance and considers the
-      // override price if the user has specified one.
-      await liquidator.liquidatePositions(currentSyntheticBalance, liquidatorOverridePrice);
-      // Check for any finished liquidations that can be withdrawn.
-      await liquidator.withdrawRewards();
-
+      await retry(
+        async () => {
+          // Update the liquidators state. This will update the clients, price feeds and gas estimator.
+          await liquidator.update();
+          // Check for liquidatable positions and submit liquidations. Bounded by current synthetic balance and
+          // considers override price if the user has specified one.
+          const currentSyntheticBalance = await syntheticToken.balanceOf(accounts[0]);
+          await liquidator.liquidatePositions(currentSyntheticBalance, liquidatorOverridePrice);
+          // Check for any finished liquidations that can be withdrawn.
+          await liquidator.withdrawRewards();
+        },
+        {
+          retries: errorRetries,
+          onRetry: error => {
+            logger.debug({
+              at: "Liquidator#index",
+              message: "An error was thrown in the execution loop - retrying",
+              error: typeof error === "string" ? new Error(error) : error
+            });
+          }
+        }
+      );
       // If the polling delay is set to 0 then the script will terminate the bot after one full run.
       if (pollingDelay === 0) {
+        logger.debug({
+          at: "Liquidator#index",
+          message: "End of severless execution loop - yielding process"
```

Comment:
```suggestion
          message: "End of serverless execution loop - yielding process"
```

Reply:
Also, is `yielding` intended to mean `stopping`?

Reply:
ye, can change the wording to something else if we want.